tests/callbacks/test_image_visualizer.py

```diff
@@ -1,112 +1,235 @@
-    )
+from tests.common.datasets import RandomImageDataset, RandomSegmentationDataset
+from tests.common.models import SimpleConvModel, SimpleSegmentationModel
+
+
+@pytest.fixture
+def test_wandb_logger(tmp_path, dummy_state):
+    pytest.importorskip('wandb', reason='wandb is optional')
+    os.environ['WANDB_DIR'] = str(tmp_path)
+    os.environ['WANDB_CACHE_DIR'] = str(Path(tmp_path) / Path('wandb_cache'))
+    os.environ['WANDB_MODE'] = 'offline'
+    dummy_state.run_name = 'wand-test-log-image'
+    logger = Logger(dummy_state, [])
+    wandb_logger = WandBLogger()
+    wandb_logger.init(dummy_state, logger)
+    return wandb_logger
+
+
+@pytest.fixture
+def comet_offline_directory(tmp_path):
+    return str(tmp_path / Path('my_cometml_runs'))
+
+
+@pytest.fixture
+def comet_logger(monkeypatch, comet_offline_directory):
+    comet_ml = pytest.importorskip('comet_ml', reason='comet_ml is optional')
+
+    monkeypatch.setattr(comet_ml, 'Experiment', comet_ml.OfflineExperiment)
+    from composer.loggers import CometMLLogger
+
+    # Set offline directory.
+    os.environ['COMET_OFFLINE_DIRECTORY'] = comet_offline_directory
+
+    comet_logger = CometMLLogger()
+    return comet_logger
+
+
+def test_image_visualizer_with_wandb(tmp_path, test_wandb_logger):
+    wandb = pytest.importorskip('wandb', reason='wandb is optional')
+
+    image_interval = 2
+    image_visualizer = ImageVisualizer(interval=f'{image_interval}ba')
+
+    dataset_size = 40
+    batch_size = 4
+    max_duration = 6
+    eval_interval = 6
+
+    trainer = Trainer(model=SimpleConvModel(),
+                      callbacks=image_visualizer,
+                      loggers=test_wandb_logger,
+                      train_dataloader=DataLoader(RandomImageDataset(size=dataset_size), batch_size),
+                      eval_dataloader=DataLoader(RandomImageDataset(size=dataset_size), batch_size),
+                      eval_interval=f'{eval_interval}ba',
+                      max_duration=f'{max_duration}ba')
 
     trainer.fit()
-    num_train_steps = int(trainer.state.timestamp.batch)
-    num_train_tables = len(in_memory_logger.data['Images/Train'])
-    num_eval_tables = len(in_memory_logger.data['Images/Eval'])
 
-    expected_train_tables = (((num_train_steps - 1) // image_visualizer.interval.value) +
-                             1) if image_visualizer.interval.unit == TimeUnit.BATCH else ((
-                                 (num_train_epochs - 1) // image_visualizer.interval.value) + 1)
+    # delete trainer to force WandBLogger to clean up in post_close
+    del trainer
+
+    expected_number_train_images = (batch_size * max_duration) / image_interval
+    expected_number_eval_images = (max_duration / eval_interval) * batch_size
 
-    assert isinstance(image_visualizer.interval, Time)
-    assert num_train_tables == expected_train_tables
-    assert num_eval_tables == num_train_epochs
+    # WandB Images are stored in cache when a WandB table is used.
```

Comment:
It looks like the segmentation tests passes without going to the cache? It would be nice if this test and the segmentation test didn't have to look in different places for the images. Is that possible, or did you determine that looking in the cache here is required?